tasks/apply-mapping/apply-mapping.yaml

```diff
@@ -62,6 +62,21 @@ spec:
             exit 0
         fi
 
+        increment_tag() {
+            local tag_prefix=$1
+            local repo=$2
+
+            existing_tags=$(oras list "$repo" --filter "${tag_prefix}-*" | grep -o "${tag_prefix}-[0-9]\+" | sort -Vr)
```

Comment:
This isn't really what I had in mind. This seems to address the most common use case, but in theory `{{ incrementer }}` could be anywhere in the tag. So, sure, `1.2-{{incrementer}}` could work now. What about `1.2.{{incrementer}}`? Or `1.{{incrementer}}.2`? Or `1-{{incrementer}}-2`? Or even `{{incrementer}}-2-3`?

I definitely think most instances of this will be using it at the end, but I don't see why not to just replace the variable with digit and search for that tag. Then it can be anywhere in the tag, regardless of hyphen, dot notation, or order. One thing I will say though is that we should not support `{{incrementer}}` multiple times in one tag, as that sounds like a nightmare

Also why are we using oras list instead of skopeo inspect? We already skopeo inspect in this task and I have never seen oras list before so I have no idea what it does (not saying it is wrong)

Reply:
Thanks for the detailed review, @johnbieren. I completely agree with your points and am implementing the necessary adjustments to address the flexible incrementer placement and use of `skopeo inspect` as recommended.